### PR TITLE
relnote(120): Zulu timezone for non-ISO `Date.parse` formats

### PR DESCRIPTION
**Pull request by vinnydiehl**
### Entry

(under the parent bullet point for the previous `Date.parse` fixes):

  - Timezone `'Z'` is now accepted for non-ISO formats (e.g. `Jan 1 1970 10:00Z`)

### Bugzilla

https://bugzilla.mozilla.org/show_bug.cgi?id=1852422

### CODE REVIEW

**Comment by hamishwillee** [comment]
1. This is cool, but we should also update https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse#non-standard_date_strings with information about this
2. If there are browser compatibility differences then it would be great to add https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse#browser_compatibility
3. Ditto for #30234

@zfox23 I assigned to you because I saw you did the previous work.

@vinnydiehl FYI You may already have had this discussion with someone, but if you see a bug that has a docs implication, can you please add the keyword `dev-doc-needed`. This allows the MDN team to pick it up when we come to do the docs for a release - we ensure that the release note, browser compatibility and docs are done are up to date at that point. I did this for this issue #30234. When we are done we change that to `dev-doc-complete` so everyone knows the work is done. We normally also cross link the bug to the MDN tracking work - for me these look something like this: https://github.com/mdn/content/issues/29778

**Comment by vinnydiehl** [comment]
> we should also update https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse#non-standard_date_strings with information about this

I've checked the patterns already in there and there are some of them already in there which are out of date.

I can certainly add information about some of the new formats which are being supported. Does it matter that some of these formats only work in pre-release versions of Firefox right now?

Thanks for the heads up on the `dev-doc-needed` tags, I've gone through and tagged all of the bugs in these last 2 PRs.

**Comment by zfox23** [comment]
Thank you for your contribution :)

> > we should also update https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse#non-standard_date_strings with information about this
> 
> I've checked the patterns already in there and there are some of them which are out of date.
> 
> I can certainly add information about some of the new formats which are being supported. Does it matter that some of these formats only work in pre-release versions of Firefox right now?

It would be _super_ helpful if you could please fix content that is out-of-date and add some of the newly-supported formats. It'd be great for you to include a comment in the JS code snippet that included data about the way the dates are parsed across browsers.

With respect to time and parsing time, keeping a BCD-like table of what happens in what browser given a specific date format is going to be a maintenance nightmare. In my opinion, the "This section contains implementation-specific behavior that can be inconsistent across implementations" note coupled with the comments in the example snippet are sufficient to keep folks on their toes about the _utter confusion_ that is the human concept of **"Time"**.

I will keep my eyes out for updates to this PR. Thanks again.

File: files/en-us/mozilla/firefox/releases/120/index.md [hunk 1]
```diff
@@ -37,12 +37,15 @@ This article provides information about the changes in Firefox 120 that affect d
...
 
   - Requirements for characters directly following numbers have been loosened to accept new formats ([Firefox bug 449921](https://bugzil.la/449921)), including:
+
     - `"DDMonYYYY"`
     - `"Mon.DD.YYYY"`
     - `"DD.Mon.YYYY"`
     - `"YYYY.MM.DD"`
     - `"Mon DD YYYY hh:mmXm"` (`am`/`pm` directly following time)
 
+  - Timezone `'Z'` is now accepted for non-ISO formats (e.g. `Jan 1 1970 10:00Z`)
```
Comment (zfox23) [inline files/en-us/mozilla/firefox/releases/120/index.md:47]: Pinging this PR , and also adding a suggestion to add the Bugzilla link:

```suggestion
  - Timezone `'Z'` is now accepted for non-ISO formats (e.g. `Jan 1 1970 10:00Z`) ([Firefox bug 1852422](https://bugzil.la/1852422))
```

@vinnydiehl, would you prefer to address the previous comments in this PR over the next couple of days pertaining to non-standard date strings, or would you prefer to merge this now and address those comments later?

**Review by zfox23 — CHANGES_REQUESTED** [review]

**Comment by vinnydiehl** [comment]
Hi Zach! Thanks for adding Bugzilla links on this and my other relnote.

Since the updates in the JS reference will not be necessarily limited to the changes in either of these relnotes, and involve existing points which are outdated, I think it'd be best to do the work in a separate PR. I've opened #30363 to track this work.

**Review by zfox23 — APPROVED** [review]

**Pull request merged**